Store/COSQLiteStorePersistentRootBackingStore.m

```diff
@@ -643,6 +659,106 @@ - (BOOL)writeItemGraph: (COItemGraph *)anItemTree
     return ok;
 }
 
+- (int64_t)rewriteRevid: (int64_t)revid
+          withItemGraph: (COItemGraph *)newItemGraph
+                version: (int64_t)newVersion
+     migratedItemGraphs: (NSDictionary *)migratedItemGraphs
+    lastBytesInDeltaRun: (int64_t)lastBytesInDeltaRun
+{
+    FMResultSet *rs = [db_ executeQuery:
+        [NSString stringWithFormat: @"SELECT parent, deltabase FROM %@ WHERE revid = ?", [self tableName]],
+        @(revid)];
+    int64_t bytesInDeltaRun = -1;
+
+    while ([rs next])
```

Comment:
Just a style nitpick, this can only return 0 or 1 rows (0 means the caller provided an invalid `revid`), the `while` loop suggests it can return several rows at first glance. Maybe just an `if` would be clearer (and report an error if 0 rows were returned?)

Reply:
Yes, this while statement doesn't make it clear there is a single result. I'm going to change it. If an invalid revid is provided, I plan to log an error and return -1. 

For now, we don't have error propagation in `COSQLiteStore` and `COSQLitePersistentRootBackingStore`, that's why I'd like to stick this style (log error and return -1 or NO). In the long run, it could be better to refactor the whole implementation to propagate detailed errors up the public API that is called.